### PR TITLE
Grenade fixes + Projectile API

### PR DESCRIPTION
**Pull request by IRacle1**
- Fixed spawning Scp2176
- Maybe fixed [elevators bug](https://discord.com/channels/656673194693885975/1061325191407489104), idk cant reproduce that

### CODE REVIEW

File: Exiled.API/Features/Pickups/Projectiles/Projectile.cs [hunk 1]
```diff
@@ -80,28 +80,41 @@ internal Projectile(ItemType type)
...
         /// Spawns a <see cref="Projectile"/>.
         /// </summary>
         /// <param name="pickup">The <see cref="Projectile"/> too spawn.</param>
         /// <param name="position">The position to spawn the <see cref="Projectile"/> at.</param>
         /// <param name="rotation">The rotation to spawn the <see cref="Projectile"/>.</param>
+        /// <param name="shouldActive">Whether the <see cref="Projectile"/> should be in active state after spawn.</param>
         /// <param name="previousOwner">An optional previous owner of the item.</param>
         /// <returns>The <see cref="Projectile"/> Spawn.</returns>
-        public static Projectile Spawn(Projectile pickup, Vector3 position, Quaternion rotation, Player previousOwner = null)
+        public static Projectile Spawn(Projectile pickup, Vector3 position, Quaternion rotation, bool shouldActive = true, Player previousOwner = null)
```
Comment (iopietro) [inline Exiled.API/Features/Pickups/Projectiles/Projectile.cs:97]:
```suggestion
        public static Projectile Spawn(Projectile pickup, Vector3 position, Quaternion rotation, bool shouldBeActive = true, Player previousOwner = null)
```

File: Exiled.API/Features/Pickups/Projectiles/Projectile.cs [hunk 2]
```diff
@@ -80,28 +80,41 @@ internal Projectile(ItemType type)
...
         {
             pickup.Position = position;
             pickup.Rotation = rotation;
             pickup.PreviousOwner = previousOwner;
             pickup.Spawn();
 
+            if (shouldActive)
```
Comment (iopietro) [inline Exiled.API/Features/Pickups/Projectiles/Projectile.cs:104]:
```suggestion
            if (shouldBeActive)
```

File: Exiled.API/Features/Pickups/Projectiles/Projectile.cs [hunk 3]
```diff
@@ -80,28 +80,41 @@ internal Projectile(ItemType type)
...
             return pickup;
         }
 
+        /// <summary>
+        /// Activates the current <see cref="Projectile"/>.
+        /// </summary>
+        public void Active()
+        {
+            Base.ServerActivate();
+        }
```
Comment (iopietro) [inline Exiled.API/Features/Pickups/Projectiles/Projectile.cs:116]:
```suggestion
        public void Active() => Base.ServerActivate();
```

File: Exiled.API/Features/Pickups/Projectiles/TimeGrenadeProjectile.cs [hunk 4]
```diff
@@ -40,36 +44,57 @@ internal TimeGrenadeProjectile(ItemType type)
...
+            set
+            {
+                if (value && Base.TargetTime == 0.0f)
+                {
+                    Base.RpcSetTime(FuseTime);
+                }
+                else if (!value && Base.TargetTime != 0.0f)
+                {
+                    Base.RpcSetTime(-Time.timeSinceLevelLoad);
+                }
```
Comment (iopietro) [inline Exiled.API/Features/Pickups/Projectiles/TimeGrenadeProjectile.cs:81]:
```suggestion
                if (value && Base.TargetTime == 0.0f)
                    Base.RpcSetTime(FuseTime);
                else if (!value && Base.TargetTime != 0.0f)
                    Base.RpcSetTime(-Time.timeSinceLevelLoad);
```

**Review by iopietro — DISMISSED** [review]
I know that those transpilers don't have so much code in them, but comment them please.

**Review by Thundermaker300 — APPROVED** [review]

**Review by iopietro — APPROVED** [review]

**Pull request merged**